znnl/data/decision_boundary.py

```diff
@@ -0,0 +1,193 @@
+"""
+ZnNL: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Data generator for decision boundary problems.
+"""
+import jax
+import jax.numpy as np
+import matplotlib.pyplot as plt
+import numpy as onp
+
+from znnl.data.data_generator import DataGenerator
+from znnl.utils.prng import PRNGKey
+
+
+def linear_boundary(data: onp.ndarray, gradient: float, intercept: float) -> np.ndarray:
+    """
+    Create a linear boundary between classes.
+
+    Parameters
+    ----------
+    data : np.ndarray (n_samples, 2)
+            Data to be converted into classes.
+    gradient : float
+            Gradient of the line, default 1.0.
+    intercept : float
+            Intercept of the line, default 1.0.
+    """
+    # y = m * x + c
+    reference_values = gradient * data[:, 0] + intercept
+
+    differences = data[:, 1] - reference_values
+
+    differences[differences > 0] = 1
+    differences[differences < 0.0] = 0
+
+    return differences
+
+
+def circle(data: onp.ndarray, radius: float = 0.25):
+    """
+    Create a circular classification problem.
+
+    For simplicity, assume the points inside the circle are
+    class 0 and outside are class 1.
+
+    Parameters
+    ----------
+    data : np.ndarray
+        Data to be converted into classes.
+    radius : float
+        Radius of the circle.
+    """
+    radii = onp.linalg.norm(data - 0.5, axis=1)
```

Comment:
This centers the data into a box with dimensions 0, 1 so that I don't need negative numbers. The data is still randomly generated. This is just a coordinate transformation.

Reply:
I just meant that this implies the data to be generated in [0, 1]. 
As this function exists outside of the class I would just like to have the information be written in the docs.

Reply:
The random data is generated inside a box of 0, 1 but this is done earlier in the random number generation call which is not accessible to the user. But I can document that.